### DISCUSSION TITLE
Tree-shaking doesn't work on re-exported modules with classes

### DISCUSSION THREAD

**Discussion by Yegorich555** [webpack/webpack#15427] [Q&A] [answered]
# Bug report

**What is the current behavior?**
If you have reexported class without usage it's not removed from production build

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Create class `export default class WontUse extends HTMLElement {
  static SomeStaticProp = "Tree-shaking doesn't work";
}
`  into wontUse.ts file
2. Create file `reexport.ts` with `export { default as WontUse } from "./wontUse";` and some default function for future usage
3. Import default function from `reexport.ts` and call it in `main.ts`
4. Build project and you can notice that WontUse class included in build-file despite it doesn't in use at all
You can check easily here: https://github.com/Yegorich555/webpack-must-have/tree/bug_treeShake
just clone and run `npm i && npm run build-prod`

**What is the expected behavior?**
All useless classes (not only functions and variables) must be excluded from build-files

**Other relevant information:**
webpack version: 5.67.0 
Node.js version: 14.19.0
Operating System: Windows 10
Additional tools: VSCode

**Comment by vankop** [comment, marked as answer]
you should set `"sideEffects"` flag in package.json https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

**Reply by Yegorich555** [reply]
Thanks, it works.

But it's difficult to control when you have external packages. In most cases you don't know what exactly files must be included in sideEffects and the single way is detect if such classes/imports are in use during the webpack-compilation. I understand that difficult to control polyfills (or extensions of global types) but checking if classes are in use or not it should be easy... 

Do you know is webpack going to tree-shake such cases automatically ? It's really important. It will help to reduce bundle-sizes in twice and more
